### ISSUE TITLE
Data source aws_ami broken in v0.7.5

### ISSUE THREAD

**Issue by darken99** [hashicorp/terraform#9326]
Hi,

It seems aws_ami resource is broken in v 0.7.5

### Terraform Configuration Files
```hcl
data "aws_ami" "ecs_ami" {
    most_recent = true

    filter {
	name = "owner-alias"
	values = ["amazon"]
    }
    filter {
	name = "name"
	values = ["amzn-ami-*-amazon-ecs-optimized*"]
    }
}
```

### Expected Behavior (Terraform 0.7.4)
AMI-ID = ami-175f1964
which is
amzn-ami-2016.03.j-amazon-ecs-optimized - ami-175f1964

### Actual Behavior (Terraform 0.7.5)
AMI-ID = ami-519a0a26
which is
amzn-ami-2014.09.1-amazon-ecs-optimized-preview3 - ami-519a0a26

### Steps to Reproduce
1. `terraform apply`

**Comment by kwilczynski** [comment]
@darken99 hi there! I am sorry that you are having issues with this resource.

This problem has been already fixed in #9277. I sincerely apologise for troubles.

Related to #9284 and #9274.

**Comment by apparentlymart** [comment]
Thanks for the bug report @darken99, and sorry for the troubles.

I'm going to close this issue in order to consolidate discussion in #9277.
